File: mobile/lib/features/task/domain/enum/task_recurrence_type.dart
enum TaskRecurrenceType { none, daily, weekly, customWeekly, monthly }

extension TaskRecurrenceTypeX on TaskRecurrenceType {
  static TaskRecurrenceType fromApi(String? value) {
    switch ((value ?? '').trim().toUpperCase()) {
      case 'DAILY':
        return TaskRecurrenceType.daily;
      case 'WEEKLY':
        return TaskRecurrenceType.weekly;
      case 'CUSTOM_WEEKLY':
      case 'WEEKLY_CUSTOM':
        return TaskRecurrenceType.customWeekly;
      case 'MONTHLY':
        return TaskRecurrenceType.monthly;
      case 'NONE':
      default:
        return TaskRecurrenceType.none;
    }
  }

  String get apiValue {
    switch (this) {
      case TaskRecurrenceType.none:
        return 'NONE';
      case TaskRecurrenceType.daily:
        return 'DAILY';
      case TaskRecurrenceType.weekly:
        return 'WEEKLY';
      case TaskRecurrenceType.customWeekly:
        return 'CUSTOM_WEEKLY';
      case TaskRecurrenceType.monthly:
        return 'MONTHLY';
    }
  }

  String get label {
    switch (this) {
      case TaskRecurrenceType.none:
        return 'No repeat';
      case TaskRecurrenceType.daily:
        return 'Daily';
      case TaskRecurrenceType.weekly:
        return 'Weekly';
      case TaskRecurrenceType.customWeekly:
        return 'Custom weekly';
      case TaskRecurrenceType.monthly:
        return 'Monthly';
    }
  }

  bool get isRecurring => this != TaskRecurrenceType.none;
}

File: backend/src/main/java/com/lifeos/backend/task/domain/enums/TaskRecurrenceType.java
package com.lifeos.backend.task.domain.enums;

/**
 * Defines how a TaskTemplate repeats.
 *
 * Important:
 * Recurrence belongs to TaskTemplate.
 * TaskInstance is the real spawned occurrence.
 */
public enum TaskRecurrenceType {

    /**
     * No recurrence.
     *
     * Used for one-time templates or manual task instances.
     */
    NONE,

    /**
     * Repeats every day.
     *
     * Example:
     * Study English every day.
     */
    DAILY,

    /**
     * Repeats weekly on the same day as recurrenceStartDate.
     *
     * Example:
     * recurrenceStartDate = Monday
     * -> repeats every Monday.
     */
    WEEKLY,

    /**
     * Repeats on selected weekdays.
     *
     * Example:
     * MONDAY,WEDNESDAY,FRIDAY
     */
    CUSTOM_WEEKLY,

    /**
     * Repeats monthly on the same day-of-month as recurrenceStartDate.
     *
     * Example:
     * recurrenceStartDate = 2026-05-08
     * -> repeats on the 8th of each month.
     */
    MONTHLY;

    public boolean isRecurring() {
        return this != NONE;
    }

    public boolean isWeeklyBased() {
        return this == WEEKLY || this == CUSTOM_WEEKLY;
    }

    public boolean requiresStartDate() {
        return this != NONE;
    }

    public boolean requiresDaysOfWeek() {
        return this == CUSTOM_WEEKLY;
    }

    public boolean usesStartDateDayOfWeek() {
        return this == WEEKLY;
    }

    public boolean usesStartDateDayOfMonth() {
        return this == MONTHLY;
    }
}